### DISCUSSION TITLE
doing non linear dimension reduction by UMAP, why using pca

### DISCUSSION THREAD

**Discussion by zonghongz** [satijalab/seurat#4511] [Q&A]
Hello Everyone
I just began to learn how to use Seurat by following the basic instruction tutorial (Guided tutorial — 2,700 PBMCs) on Seurat web site.
I followed the tutorial step by step, however I was confused when I reach step, "Run non-linear dimensional reduction (UMAP/tSNE)". The tutorial instructs to do "pbmc <- RunUMAP(pbmc, dims = 1:10)". However, by checking the help doc for RunUNMAP, I found that dimensional reduction to use for the RunUMAP() is PCA. My understanding is that UMAP is independent non linear dimension reduction method whereas PCA is linear dimension reduction method, why Seurat use PCA for RunUMAP function here? 

I would appreciate if anyone may clarify my understanding here.

Thanks

Zong

**Comment by saketkc** [comment]
Your understanding is correct. The PCA input in this case is a "denoised" version of the data, so improves cluster separation overall when visualized through UMAP.

**Reply by mea2712** [reply]
How would you go about clustering on UMAP instead of PCA to pick on some non-linear effects?
Can we RunUMAP(...,n.components=50L) and then FindClusters(..., reduction='umap')? What would be the best way to plot this data? Just the 2 first UMAP components or RunTSNE()?
Thanks!

**Reply by saketkc** [reply]
Clustering is done by constructing a KNN graph based on euclidean distance in the PCA space. UMAP is for visualization.

**Reply by mea2712** [reply]
Yes indeed. What I meant was to construct the graph on the UMAP dimension and then cluster -- FindNeighbors(..., reduction='umap', annoy.metric='cosine'). Do any insights on that? Thanks!

**Reply by zonghongz** [reply]
Thanks saketkc for your reply. I also have the same question as mea2712. Thanks

**Reply by flde** [reply]
I would also be interested in using the RunUMAP output knn for FindCluster. There is no return value for the graph in RunUMAP yet, right? However, the uwot method of RunUMAP uses the annoy method to compute knn. Is that the same graph as computing the knn with FindNeighbors method annoy? Very thankful for comments to clean up some confusion of mine.
